demonstrations/tutorial_backprop.py

```diff
@@ -0,0 +1,431 @@
+r"""
+Quantum gradients with backpropagation
+======================================
+
+.. meta::
+    :property="og:description": Compare and contrast the parameter-shift rule with backpropagation.
+    :property="og:image": https://pennylane.ai/qml/_images/sphx_glr_tutorial_backprop_002.png
+
+In PennyLane, any quantum device, whether a hardware device or a simulator, can be
+trained using the :doc:`parameter-shift rule </glossary/parameter_shift>` to compute quantum
+gradients. Indeed, the parameter-shift rule is ideally suited to hardware devices, as it does
+not require any knowledge about the internal workings of the device; it is sufficient to treat
+the device as a 'black box', and to query it with different input values in order to determine the gradient.
+
+When working with simulators, however, we *do* have access to the internal (classical)
+computations being performed. This allows us to take advantage of other methods of computing the
+gradient, such as backpropagation, which may be advantageous in certain regimes. In this tutorial,
+we will compare and contrast the parameter-shift rule against backpropagation, using
+the PennyLane :class:`default.qubit.tf <pennylane.plugins.default_qubit_tf.DefaultQubitTF>`
+device.
+
+The parameter-shift rule
+------------------------
+
+The parameter-shift rules states that, given a variational quantum circuit :math:`U(\boldsymbol
+\theta)` composed of parametrized Pauli rotations, and some measured observable :math:`\hat{B}`, the
+derivative of the expectation value
+
+.. math::
+
+	\langle \hat{B} \rangle (\boldsymbol\theta) =
+	\langle 0 \mid U(\boldsymbol\theta)^\dagger \hat{B} U(\boldsymbol\theta) \mid 0\rangle

```

Comment:
```suggestion
	\langle 0 \vert U(\boldsymbol\theta)^\dagger \hat{B} U(\boldsymbol\theta) \vert 0\rangle
```
"`\mid`" causes weird extra spaces: 
![image](https://user-images.githubusercontent.com/7213358/89416472-4c180780-d6fb-11ea-81af-6ec9dccc5d00.png)